ee/app/livechat-enterprise/server/hooks/scheduleAutoTransfer.ts

```diff
@@ -0,0 +1,49 @@
+import { AutoTransferChatScheduler } from '../lib/AutoTransferChatScheduler';
+import { callbacks } from '../../../../../app/callbacks/server';
+import { settings } from '../../../../../app/settings/server';
+import { LivechatRooms } from '../../../../../app/models/server';
+
+
+const handleAfterTakeInquiryCallback = async (inquiry: any = {}): Promise<any> => {
+	const { rid } = inquiry;
+	if (!rid || !rid.trim()) {
+		return;
+	}
+
+	const timeout = settings.get('Livechat_auto_transfer_chat_timeout');
+	if (!timeout || timeout <= 0) {
+		return inquiry;
+	}
+
+	const room = LivechatRooms.findOneById(rid, { autoTransferredAt: 1 });
+	if (!room || room.autoTransferredAt) {
+		return inquiry;
+	}
+
+	await AutoTransferChatScheduler.scheduleRoom(rid, timeout as number);
+
+	return inquiry;
+};
+
+const handleAfterSaveMessage = async (message: any = {}, room: any = {}): Promise<any> => {
+	const { _id: rid, t } = room;
+	const { token } = message;
+
+	if (!rid || !message || rid === '' || t !== 'l' || token) {
+		return;
+	}
+
+	await AutoTransferChatScheduler.unscheduleRoom(rid);
```

Comment:
This method is being called after all agent messages, which is not necessary.
The method must be called only when the auto-transfer process is scheduled.

Reply:
got it, I've added the following check there. now it won't get called if the setting value is 0 or in other words the auto-transfer feature is off
```
	const timeout = settings.get('Livechat_auto_transfer_chat_timeout');
	if (!timeout || timeout <= 0) {
		return;
	}
```